<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Package to decouple gene sets from statistics â€¢ decoupleR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Package to decouple gene sets from statistics">
<meta property="og:description" content="Transcriptome profiling followed by differential gene
    expression analysis often leads to lists of genes that are hard to
    analyze and interpret. Downstream analysis tools can be used to
    summarize deregulation events into a smaller set of biologically
    interpretable features.  In particular, methods that estimate the
    activity of transcription factors (TFs) from gene expression are
    commonly used. It has been shown that the transcriptional targets of a
    TF yield a much more robust estimation of the TF activity than
    observing the expression of the TF itself. Consequently, for the
    estimation of transcription factor activities, a network of
    transcriptional regulation is required in combination with a
    statistical algorithm that summarizes the expression of the target
    genes into a single activity score. Over the years, many different
    regulatory networks and statistical algorithms have been developed,
    mostly in a fixed combination of one network and one algorithm. To
    systematically evaluate both networks and algorithms, we developed
    decoupleR , an R package that allows users to apply efficiently any
    combination provided.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-119440867-11"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-119440867-11');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">decoupleR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="articles/decoupleR.html">Get started</a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/saezlab/decoupleR/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://twitter.com/saezlab">
    <span class="fab fa-twitter fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="http://www.saezlab.org">
    <span class="fas fa-university"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div id="decoupler" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#decoupler" class="anchor"></a>decoupleR</h1></div>
<!-- badges: start -->

<div id="overview" class="section level2">
<h2 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h2>
<p>Transcriptome profiling followed by differential gene expression analysis often leads to lists of genes that are hard to analyze and interpret. Downstream analysis tools can be used to summarize deregulation events into a smaller set of biologically interpretable features. In particular, methods that estimate the activity of transcription factors (TFs) from gene expression are commonly used. It has been shown that the transcriptional targets of a TF yield a much more robust estimation of the TF activity than observing the expression of the TF itself. Consequently, for the estimation of transcription factor activities, a network of transcriptional regulation is required in combination with a statistical algorithm that summarizes the expression of the target genes into a single activity score. Over the years, many different regulatory networks and statistical algorithms have been developed, mostly in a fixed combination of one network and one algorithm. To systematically evaluate both networks and algorithms, we developed decoupleR , an R package that allows users to apply efficiently any combination provided.</p>
<p>For more information about how this package has been used with real data, please check the following links:</p>
<ul>
<li>
<a href="https://docs.google.com/document/d/1aYD3IF1tj7H6vUFqKnfHB6d6Trs_oit7umjk5mQDNZ0/edit">Decoupling statistics and networks: a crowdsourced systematic assessment of transcription factor activity estimation from transcriptomics data</a>.</li>
<li>
<a href="https://github.com/saezlab/decoupleR/tree/benchmark">Creation of benchmarking pipelines</a>.</li>
</ul>
</div>
<div id="installation-instructions" class="section level2">
<h2 class="hasAnchor">
<a href="#installation-instructions" class="anchor"></a>Installation instructions</h2>
<p>Get the latest stable <code>R</code> release from <a href="http://cran.r-project.org/">CRAN</a>.</p>
<!-- Then install `decoupleR` using from [Bioconductor](http://bioconductor.org/) the following code: -->
<!-- ```{r bioconductor_install, eval = FALSE} -->
<!-- if (!requireNamespace("BiocManager", quietly = TRUE)) { -->
<!--     install.packages("BiocManager") -->
<!-- } -->
<!-- BiocManager::install("decoupleR") -->
<!-- ``` -->
<p>Then install development version from <a href="https://github.com/">GitHub</a> with:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># install.packages("devtools")</span>
<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"saezlab/decoupleR"</span><span class="op">)</span></code></pre></div>
</div>
<div id="usage" class="section level2">
<h2 class="hasAnchor">
<a href="#usage" class="anchor"></a>Usage</h2>
<div id="load-packaga-and-data" class="section level3">
<h3 class="hasAnchor">
<a href="#load-packaga-and-data" class="anchor"></a>Load packaga and data</h3>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://saezlab.github.io/decoupleR/">decoupleR</a></span><span class="op">)</span>

<span class="va">inputs_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"testdata"</span>, <span class="st">"inputs"</span>, package <span class="op">=</span> <span class="st">"decoupleR"</span><span class="op">)</span>

<span class="va">mat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">inputs_dir</span>, <span class="st">"input-expr_matrix.rds"</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/glimpse.html">glimpse</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt;  num [1:18490, 1:4] 3.251 0.283 -2.253 0.782 -4.575 ...</span>
<span class="co">#&gt;  - attr(*, "dimnames")=List of 2</span>
<span class="co">#&gt;   ..$ : chr [1:18490] "A1BG" "A1CF" "A2M" "A2ML1" ...</span>
<span class="co">#&gt;   ..$ : chr [1:4] "GSM2753335" "GSM2753336" "GSM2753337" "GSM2753338"</span>

<span class="va">network</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">inputs_dir</span>, <span class="st">"input-dorothea_genesets.rds"</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/glimpse.html">glimpse</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; Rows: 151</span>
<span class="co">#&gt; Columns: 5</span>
<span class="co">#&gt; $ tf         &lt;chr&gt; "FOXO4", "FOXO4", "FOXO4", "FOXO4", "FOXO4", "FOXO4", "FOXOâ€¦</span>
<span class="co">#&gt; $ confidence &lt;chr&gt; "A", "A", "A", "A", "A", "A", "A", "A", "A", "A", "A", "A",â€¦</span>
<span class="co">#&gt; $ target     &lt;chr&gt; "BCL2L11", "BCL6", "CDKN1A", "CDKN1B", "G6PC", "GADD45A", "â€¦</span>
<span class="co">#&gt; $ mor        &lt;dbl&gt; 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,â€¦</span>
<span class="co">#&gt; $ likelihood &lt;dbl&gt; 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,â€¦</span></code></pre></div>
</div>
<div id="decouple-wrapper" class="section level3">
<h3 class="hasAnchor">
<a href="#decouple-wrapper" class="anchor"></a>Decouple wrapper</h3>
<p><code><a href="reference/decouple.html">decouple()</a></code> allows access to all <strong>decoupleR</strong> available statistics in one place. Statistic functions inside <strong>decoupleR</strong> always return a tidy tibble that can be easily processed with the tools provide by the <a href="https://www.tidyverse.org/">tidyverse ecosystem</a>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="reference/decouple.html">decouple</a></span><span class="op">(</span>
    mat <span class="op">=</span> <span class="va">mat</span>,
    network <span class="op">=</span> <span class="va">network</span>,
    .source <span class="op">=</span> <span class="st">"tf"</span>,
    .target <span class="op">=</span> <span class="st">"target"</span>,
    statistics <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"gsva"</span>, <span class="st">"mean"</span>, <span class="st">"pscira"</span>, <span class="st">"scira"</span>, <span class="st">"viper"</span>, <span class="st">"ora"</span><span class="op">)</span>,
    args <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>
        gsva <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>,
        mean <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>.mor <span class="op">=</span> <span class="st">"mor"</span>, .likelihood <span class="op">=</span> <span class="st">"likelihood"</span><span class="op">)</span>,
        pscira <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>.mor <span class="op">=</span> <span class="st">"mor"</span><span class="op">)</span>,
        scira <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>.mor <span class="op">=</span> <span class="st">"mor"</span><span class="op">)</span>,
        viper <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>.mor <span class="op">=</span> <span class="st">"mor"</span>, .likelihood <span class="op">=</span> <span class="st">"likelihood"</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>,
        ora <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="op">)</span>
    <span class="op">)</span>
<span class="op">)</span>
<span class="co">#&gt; # A tibble: 140 x 11</span>
<span class="co">#&gt;    run_id statistic tf    condition    score p_value estimate conf.low conf.high</span>
<span class="co">#&gt;    &lt;chr&gt;  &lt;chr&gt;     &lt;chr&gt; &lt;chr&gt;        &lt;dbl&gt;   &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt;     &lt;dbl&gt;</span>
<span class="co">#&gt;  1 1      gsva      FOXO4 GSM2753335 -0.380       NA       NA       NA        NA</span>
<span class="co">#&gt;  2 1      gsva      FOXO4 GSM2753336 -0.300       NA       NA       NA        NA</span>
<span class="co">#&gt;  3 1      gsva      FOXO4 GSM2753337  0.239       NA       NA       NA        NA</span>
<span class="co">#&gt;  4 1      gsva      FOXO4 GSM2753338  0.0907      NA       NA       NA        NA</span>
<span class="co">#&gt;  5 1      gsva      NFIC  GSM2753335 -0.0845      NA       NA       NA        NA</span>
<span class="co">#&gt;  6 1      gsva      NFIC  GSM2753336  0.0778      NA       NA       NA        NA</span>
<span class="co">#&gt;  7 1      gsva      NFIC  GSM2753337 -0.260       NA       NA       NA        NA</span>
<span class="co">#&gt;  8 1      gsva      NFIC  GSM2753338  0.281       NA       NA       NA        NA</span>
<span class="co">#&gt;  9 1      gsva      RFXAP GSM2753335 -0.810       NA       NA       NA        NA</span>
<span class="co">#&gt; 10 1      gsva      RFXAP GSM2753336 -0.472       NA       NA       NA        NA</span>
<span class="co">#&gt; # â€¦ with 130 more rows, and 2 more variables: method &lt;chr&gt;, alternative &lt;chr&gt;</span></code></pre></div>
</div>
<div id="individual-parts" class="section level3">
<h3 class="hasAnchor">
<a href="#individual-parts" class="anchor"></a>Individual parts</h3>
<p>In turn, we recognize that the use of individual statistics may be of interest. Therefore, these are also exported and ready for use. All statistics follow the same design pattern and arguments, so moving between statistics could be very comfortable.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># viper call is equivalent to the one made by decouple() above.</span>
<span class="fu"><a href="reference/run_viper.html">run_viper</a></span><span class="op">(</span>
    mat <span class="op">=</span> <span class="va">mat</span>,
    network <span class="op">=</span> <span class="va">network</span>,
    .source <span class="op">=</span> <span class="st">"tf"</span>,
    .target <span class="op">=</span> <span class="st">"target"</span>,
    .likelihood <span class="op">=</span> <span class="st">"likelihood"</span>,
    verbose <span class="op">=</span> <span class="cn">FALSE</span>
<span class="op">)</span>
<span class="co">#&gt; # A tibble: 20 x 4</span>
<span class="co">#&gt;    statistic tf     condition    score</span>
<span class="co">#&gt;    &lt;chr&gt;     &lt;chr&gt;  &lt;chr&gt;        &lt;dbl&gt;</span>
<span class="co">#&gt;  1 viper     FOXO4  GSM2753335  1.34  </span>
<span class="co">#&gt;  2 viper     FOXO4  GSM2753336  1.18  </span>
<span class="co">#&gt;  3 viper     FOXO4  GSM2753337  1.44  </span>
<span class="co">#&gt;  4 viper     FOXO4  GSM2753338  1.21  </span>
<span class="co">#&gt;  5 viper     NFIC   GSM2753335  0.0696</span>
<span class="co">#&gt;  6 viper     NFIC   GSM2753336 -0.0265</span>
<span class="co">#&gt;  7 viper     NFIC   GSM2753337 -0.516 </span>
<span class="co">#&gt;  8 viper     NFIC   GSM2753338 -0.543 </span>
<span class="co">#&gt;  9 viper     RFXAP  GSM2753335  0.488 </span>
<span class="co">#&gt; 10 viper     RFXAP  GSM2753336  1.32  </span>
<span class="co">#&gt; 11 viper     RFXAP  GSM2753337  1.93  </span>
<span class="co">#&gt; 12 viper     RFXAP  GSM2753338  1.93  </span>
<span class="co">#&gt; 13 viper     SMAD3  GSM2753335  0.176 </span>
<span class="co">#&gt; 14 viper     SMAD3  GSM2753336  0.0426</span>
<span class="co">#&gt; 15 viper     SMAD3  GSM2753337  0.219 </span>
<span class="co">#&gt; 16 viper     SMAD3  GSM2753338  0.142 </span>
<span class="co">#&gt; 17 viper     TFAP2A GSM2753335  0.722 </span>
<span class="co">#&gt; 18 viper     TFAP2A GSM2753336  0.582 </span>
<span class="co">#&gt; 19 viper     TFAP2A GSM2753337  0.462 </span>
<span class="co">#&gt; 20 viper     TFAP2A GSM2753338  0.330</span></code></pre></div>
<!-- ## Citation -->
<!-- Below is the citation output from using `citation('decoupleR')` in R. Please -->
<!-- run this yourself to check for any updates on how to cite __decoupleR__. -->
<!-- ```{r 'citation', eval = requireNamespace('decoupleR')} -->
<!-- print(citation("decoupleR"), bibtex = TRUE) -->
<!-- ``` -->
<!-- Please note that the `decoupleR` was only made possible thanks to many other R -->
<!-- and bioinformatics software authors, which are cited either in the vignettes -->
<!-- and/or the paper(s) describing this package. -->
</div>
</div>
<div id="contributing-to-decoupler" class="section level2">
<h2 class="hasAnchor">
<a href="#contributing-to-decoupler" class="anchor"></a>Contributing to decoupleR</h2>
<p>Are you interested in adding a new statistical method or collaborating in the development of internal tools that allow the extension of the package? Please check out our <a href="https://saezlab.github.io/decoupleR/CONTRIBUTING.html">contribution guide</a>.</p>
<hr>
<p>Please note that this project is released with a <a href="https://saezlab.github.io/decoupleR/CODE_OF_CONDUCT">Contributor Code of Conduct</a>. By participating in this project you agree to abide by its terms.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Download from BIOC at <br><a href="https://www.bioconductor.org/packages/decoupleR">https://â€‹www.bioconductor.org/â€‹packages/â€‹decoupleR</a>
</li>
<li>Browse source code at <br><a href="https://github.com/saezlab/decoupleR/">https://â€‹github.com/â€‹saezlab/â€‹decoupleR/â€‹</a>
</li>
<li>Report a bug at <br><a href="https://github.com/saezlab/decoupleR/issues">https://â€‹github.com/â€‹saezlab/â€‹decoupleR/â€‹issues</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></li>
</ul>
</div>
<div class="community">
<h2>Community</h2>
<ul class="list-unstyled">
<li><a href="CONTRIBUTING.html">Contributing guide</a></li>
<li><a href="CODE_OF_CONDUCT.html">Code of conduct</a></li>
</ul>
</div>
<div class="citation">
<h2>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html">Citing decoupleR</a></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>JesÃºs VÃ©lez <br><small class="roles"> Maintainer, author </small> <a href="https://orcid.org/0000-0001-5128-3838" target="orcid.widget" aria-label="ORCID"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Christian H. Holland <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-3060-5786" target="orcid.widget" aria-label="ORCID"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

  <div class="dev-status">
<h2>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://www.tidyverse.org/lifecycle/#experimental"><img src="https://img.shields.io/badge/lifecycle-experimental-orange.svg" alt="Lifecycle:experimental"></a></li>
<li><a href="https://github.com/saezlab/decoupleR/actions"><img src="https://github.com/saezlab/decoupleR/workflows/R-CMD-check-bioc/badge.svg" alt="R build status"></a></li>
<li><a href="https://codecov.io/gh/saezlab/decoupleR?branch=master"><img src="https://codecov.io/gh/saezlab/decoupleR/branch/master/graph/badge.svg" alt="Codecov test coverage"></a></li>
<li><a href="https://github.com/saezlab/decoupleR/issues"><img src="https://img.shields.io/github/issues/saezlab/decoupleR" alt="GitHub issues"></a></li>
</ul>
</div>
</div>
</div>


      <footer><div class="copyright">
  <p>Developed by JesÃºs VÃ©lez, Christian H. Holland.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
