---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

# decoupleR

<!-- badges: start -->
[![Build Status](https://travis-ci.com/saezlab/decoupleR.svg?token=PagY1pyvMyyL3AJHRy5V&branch=master)](https://travis-ci.com/saezlab/decoupleR)
[![Lifecycle:experimental](https://img.shields.io/badge/lifecycle-experimental-orange.svg)](https://www.tidyverse.org/lifecycle/#experimental)
[![Codecov test coverage](https://codecov.io/gh/saezlab/decoupleR/branch/master/graph/badge.svg)](https://codecov.io/gh/saezlab/decoupleR?branch=master)
<!-- badges: end -->

> a community effort by [saezlab](http://saezlab.org) members

## Overview

Transcriptome profiling followed by differential gene expression analysis often leads to lists of genes that are hard to analyze and interpret. Downstream analysis tools can be used to summarize deregulation events into a smaller set of biologically interpretable features. In particular, methods that estimate the activity of transcription factors (TFs) from gene expression are commonly used. It has been shown that the transcriptional targets of a TF yield a much more robust estimation of the TF activity than observing the expression of the TF itself.  Consequently, for the estimation of transcription factor activities, a network of transcriptional regulation is required in combination with a statistical algorithm that summarizes the expression of the target genes into a single activity score. Over the years, many different regulatory networks and statistical algorithms have been developed, mostly in a fixed combination of one network and one algorithm. To systematically evaluate both networks and algorithms, we developed decoupleR , an R package that allows users to apply efficiently any combination provided. 

As an initial set of regulatory networks, we integrated the following resources:

* [DoRothEA](https://github.com/saezlab/dorothea)
* [CHEA3](https://amp.pharm.mssm.edu/ChEA3)
* [RegNetwork](http://www.regnetworkweb.org/)

And the following TF activity inference methods:

### Estimation based on enrichment of sets on rankings

* [viper](http://bioconductor.org/packages/release/bioc/html/viper.html)
* GSEA as implemented in [fgsea](https://www.bioconductor.org/packages/release/bioc/html/fgsea.html)
* [GSVA](https://www.bioconductor.org/packages/release/bioc/html/GSVA.html)

### Estimation from linear models

* SCIRA: Linear models to estimate TF activities from gene expression as defined [here](https://www.nature.com/articles/s41525-020-00151-y?elqTrackId=d7efb03cf5174fe2ba84e1c34d602b13)

* pscira: Linear combination of gene expression based on mode of regulation followed by a comparison to a random null model.

### Estimation from statistics

* mean: Weighted mean that allows the use of directions and contribution weights.

* normalized_mean: Similar as above, however the final score is corrected based on a null (random) model   We benchmarked an initial set of 84 combinations, comprising 7 methods and 13 networks. 

We evaluated the precision of different combinations in recovering perturbed TFs from different collections of gene expression datasets. Additionally, we tested the effects of combining multiple sources and estimations. We set up the package in a modular way which makes it easy and intuitive to extend it with further statistics or networks. We invite the community to participate by implementing their own statistics or integrating their gene regulatory network. With the decoupleR package, we lay the foundation for a crowdsourced systematic assessment of transcription factor activity estimation from transcriptomics data.

## How to contribute?

### How to add a new statistic

* Open an [issue](https://github.com/saezlab/decoupleR/issues) stating which 
statistic you would like to add. Assign it to yourself and label it with the 
badge `enhancement`. Please check before if not someone else is already working 
on implementing this statistic.

* Don't work on the `master branch`. Either create a new `branch` with a 
meaningful name or make a `fork`. When the function is implemented and tested 
(!) we will use `pull requests` to integrate the new feature in the 
`master branch`.

* Assuming you would like to implement `GSEA`. You will need to define the 
following set of functions:
    * All following functions will be written in the script `R/gsea.R`.
    * `run_gsea(emat, genesets, list=options(), gs_resource, tidy)`. Please 
    check the function 
    [run_viper](https://github.com/saezlab/decoupleR/blob/master/R/viper.R#L30) to 
    understand the arguments.
    * `make_gsea_genesets(genesets)`. First define a standardized input for gene
    sets in *tibble/dataframe* format for your statistic. Then this function should 
    convert this table in the required input for the underlying function (in this 
    case list of gene sets for the function `fgsea()`). Check an example 
    [here](https://github.com/saezlab/decoupleR/blob/master/R/viper.R#L68). In 
    the roxygen comments of the function you define which columns must be 
    available in the standardized format. 
    * In parallel you need to define helper functions that convert every 
    available gene set (e.g. progeny and dorothea gene sets) to your defined 
    standardized input (e.g. `progeny2gsea()`). Check an example 
    [here](https://github.com/saezlab/decoupleR/blob/master/R/viper.R#L90)
    * **All functions must be documented following the roxygen2 standard.**
    
* We will use unit tests for the `run_x()` functions to ensure that our 
functions work properly. [Here](https://github.com/saezlab/decoupleR/blob/master/tests/testthat/test-viper.R) are 
examples for the `run_viper` function. 

* *Optional*: For a consistent coding style and efficient implementation we will 
mainly use `tidyverse`.

### How to add a new gene set resource
* Open an [issue](https://github.com/saezlab/decoupleR/issues) stating which 
gene set resource you would like to add. Assign it to yourself and label it with
the badge `enhancement`. Please check before if not someone else is already 
working on implementing this gene set resource.

* Don't work on the `master branch`. Either create a new `branch` with a 
meaningful name or make a `fork`. When the function is implemented and tested 
(!) we will use `pull requests` to integrate the new feature in the 
`master branch`.

* Define helper functions for each available statistics that convert your gene 
sets to the standardized format of the respective statistic 
(e.g. `your_genesets2viper()`).

* Deposit a representative selection of gene sets in the directory 
`inst/testdata` and implement unit tests for all available statistics coupled 
with your new gene sets.

### How to integrate your changes in the master branch
Before you push your changes to your development branch please first make sure that `devtools::check()` runs without any problems on your local machine (warning(s) might be acceptable, but will be decided on a case-by-case basis). After you pushed then to your development branch [travis](https://travis-ci.com/github/saezlab/decoupleR) will also build (`R CMD build`) and check (`R CMD check`) the package. This might take up to 15 minutes so better be sure it passes at the first try. **Your changes will only be integrated in the master branch when the package passes all checks on travis.**

If all requirements are fulfilled (passed `R CMD check` on your local machine and travis, implemented unit tests, proper documentation), add a brief and concise summary of the implemented features in the [NEWS.md](https://github.com/saezlab/decoupleR/blob/master/NEWS.md) file and bump the version in the [DESCRIPTION](https://github.com/saezlab/decoupleR/blob/master/DESCRIPTION) file. In addition list you as contributor (ctr) and/or author (aut) also in the [DESCRIPTION](https://github.com/saezlab/decoupleR/blob/master/DESCRIPTION) file.

After you open a pull request please refer in the commit message to the related issue, e.g. `Closes #5`. This message automatically closes issue 5, after successful integration. Also request review from someone else to double check your code. **Do not accept your own pull request without someone else having checked your code**.

## Installation
```r
# install the development version from GitHub
# install.packages("remotes")
remotes::install_github("saezlab/decoupleR")
```
