<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to decoupleR • decoupleR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to decoupleR">
<meta property="og:description" content="decoupleR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-119440867-11"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-119440867-11');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">decoupleR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.1.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/decoupleR.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/saezlab/decoupleR/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://twitter.com/saezlab">
    <span class="fab fa-twitter fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="http://www.saezlab.org">
    <span class="fas fa-university"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="decoupleR_files/header-attrs-2.11/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Introduction to decoupleR</h1>
                        <h4 class="author">Jesús Vélez</h4>
            <address class="author_afil">
      National Autonomous University of Mexico<br><a class="author_email" href="mailto:#"></a><a href="mailto:jvelezmagic@gmail.com" class="email">jvelezmagic@gmail.com</a>
      </address>
                  
      
      <small class="dont-index">Source: <a href="https://github.com/saezlab/decoupleR/blob/master/vignettes/decoupleR.Rmd"><code>vignettes/decoupleR.Rmd</code></a></small>
      <div class="hidden name"><code>decoupleR.Rmd</code></div>

    </div>

    
    
<div id="basics" class="section level1">
<h1 class="hasAnchor">
<a href="#basics" class="anchor"></a>Basics</h1>
<div id="install-decoupler" class="section level2">
<h2 class="hasAnchor">
<a href="#install-decoupler" class="anchor"></a>Install <code>decoupleR</code>
</h2>
<p><code>R</code> is an open-source statistical environment which can be easily modified to enhance its functionality via packages. <em><a href="https://bioconductor.org/packages/3.13/decoupleR">decoupleR</a></em> is an <code>R</code> package available via the <a href="http://bioconductor.org">Bioconductor</a> repository for packages. <code>R</code> can be installed on any operating system from <a href="https://cran.r-project.org/">CRAN</a> after which you can install <em><a href="https://bioconductor.org/packages/3.13/decoupleR">decoupleR</a></em> by using the following commands in your <code>R</code> session:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>
<span class="op">}</span>

<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">"decoupleR"</span><span class="op">)</span>

<span class="co"># Check that you have a valid Bioconductor installation</span>
<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/valid.html">valid</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p>You can install the development version from <a href="https://github.com/">GitHub</a> with:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">"saezlab/decoupleR"</span><span class="op">)</span></code></pre></div>
</div>
<div id="required-knowledge" class="section level2">
<h2 class="hasAnchor">
<a href="#required-knowledge" class="anchor"></a>Required knowledge</h2>
<p><em><a href="https://bioconductor.org/packages/3.13/decoupleR">decoupleR</a></em> is based on many other packages and in particular in those that have implemented the infrastructure needed for dealing with functional analysis. That is, packages like <em><a href="https://bioconductor.org/packages/3.13/viper">viper</a></em> or <em><a href="https://bioconductor.org/packages/3.13/GSVA">GSVA</a></em>, among others. This in order to have a centralized place from which to apply different statistics to the same data set without the need and work that would require testing in isolation. Opening the possibility of developing benchmarks that can grow easily.</p>
</div>
<div id="asking-for-help" class="section level2">
<h2 class="hasAnchor">
<a href="#asking-for-help" class="anchor"></a>Asking for help</h2>
<p>As package developers, we try to explain clearly how to use our packages and in which order to use the functions. But <code>R</code> and <code>Bioconductor</code> have a steep learning curve so it is critical to learn where to ask for help. We would like to highlight the <a href="https://support.bioconductor.org/">Bioconductor support site</a> as the main resource for getting help: remember to use the <code>decoupleR</code> tag and check <a href="https://support.bioconductor.org/t/decoupleR/">the older posts</a>. Other alternatives are available such as creating GitHub issues and tweeting. However, please note that if you want to receive help you should adhere to the <a href="http://www.bioconductor.org/help/support/posting-guide/">posting guidelines</a>. It is particularly critical that you provide a small reproducible example and your session information so package developers can track down the source of the error.</p>
</div>
<div id="citing-decoupler" class="section level2">
<h2 class="hasAnchor">
<a href="#citing-decoupler" class="anchor"></a>Citing <code>decoupleR</code>
</h2>
<p>We hope that <em><a href="https://bioconductor.org/packages/3.13/decoupleR">decoupleR</a></em> will be useful for your research. Please use the following information to cite the package and the overall approach. Thank you!</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/citation.html">citation</a></span><span class="op">(</span><span class="st">"decoupleR"</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; To cite decoupleR in publications, please use:</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;   Badia-i-Mompel P., Vélez J., Braunger J., Geiss C., Dimitrov D.,</span>
<span class="co">#&gt;   Müller-Dott S., Taus P., Dugourd A., Holland C.H., Ramirez Flores</span>
<span class="co">#&gt;   R.O.  and Saez-Rodriguez J. 2021. decoupleR: Ensemble of</span>
<span class="co">#&gt;   computational methods to infer biological activities from omics data.</span>
<span class="co">#&gt;   bioRxiv.  https://doi.org/10.1101/2021.11.04.46727</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; A BibTeX entry for LaTeX users is</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;   @Article{,</span>
<span class="co">#&gt;     author = {Pau Badia-i-Mompel and Jesús Vélez and Jana Braunger and Celina Geiss and Daniel Dimitrov and Sophia Müller-Dott and Petr Taus and Aurelien Dugourd and Christian H. Holland and Ricardo O. Ramirez Flores and Julio Saez-Rodriguez},</span>
<span class="co">#&gt;     title = {decoupleR: Ensemble of computational methods to infer biological activities from omics data},</span>
<span class="co">#&gt;     journal = {bioRxiv},</span>
<span class="co">#&gt;     year = {2021},</span>
<span class="co">#&gt;     doi = {https://doi.org/10.1101/2021.11.04.467271},</span>
<span class="co">#&gt;   }</span></code></pre></div>
</div>
</div>
<div id="quick-start-to-using-to-decoupler" class="section level1">
<h1 class="hasAnchor">
<a href="#quick-start-to-using-to-decoupler" class="anchor"></a>Quick start to using to <code>decoupleR</code>
</h1>
<div id="libraries" class="section level2">
<h2 class="hasAnchor">
<a href="#libraries" class="anchor"></a>Libraries</h2>
<p><em><a href="https://bioconductor.org/packages/3.13/decoupleR">decoupleR</a></em> provides different statistics to calculate the regulatory activity given a <code>matrix</code> of molecular readouts and a <code>network</code>. It incorporates pre-existing methods to avoid recreating the wheel while implementing its own methods under an evaluation standard. Therefore, it provides flexibility when evaluating a data set with different statistics.</p>
<p>Since inputs and outputs are always tibbles (i.e. special data frames), incorporating <em><a href="https://github.com/tidyverse/dplyr">dplyr</a></em> into your workflow can be useful for manipulating results, but it is not necessary.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://saezlab.github.io/decoupleR/">decoupleR</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span></code></pre></div>
</div>
<div id="input-data" class="section level2">
<h2 class="hasAnchor">
<a href="#input-data" class="anchor"></a>Input data</h2>
<p>In order to use it, you first need to have a <code>matrix</code> where the rows represent the target nodes and the columns the samples. In addition, it is necessary to provide a prior knowledge <code>network</code> that contains at least two columns corresponding to the source and target nodes. It is noteworthy that certain methods will require specifying additional metadata columns. For instance, the mode of regulation (MoR) or the likelihood of the interaction.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">inputs_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"testdata"</span>, <span class="st">"inputs"</span>, package <span class="op">=</span> <span class="st">"decoupleR"</span><span class="op">)</span>

<span class="va">mat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">inputs_dir</span>, <span class="st">"input-expr_matrix.rds"</span><span class="op">)</span> <span class="op"><a href="../reference/%&gt;%.html">%&gt;%</a></span>
    <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="../reference/%&gt;%.html">%&gt;%</a></span>
    <span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html">glimpse</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt;  num [1:18490, 1:4] 3.251 0.283 -2.253 0.782 -4.575 ...</span>
<span class="co">#&gt;  - attr(*, "dimnames")=List of 2</span>
<span class="co">#&gt;   ..$ : chr [1:18490] "A1BG" "A1CF" "A2M" "A2ML1" ...</span>
<span class="co">#&gt;   ..$ : chr [1:4] "GSM2753335" "GSM2753336" "GSM2753337" "GSM2753338"</span>

<span class="va">network</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">inputs_dir</span>, <span class="st">"input-dorothea_genesets.rds"</span><span class="op">)</span> <span class="op"><a href="../reference/%&gt;%.html">%&gt;%</a></span>
    <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="../reference/%&gt;%.html">%&gt;%</a></span>
    <span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html">glimpse</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; Rows: 151</span>
<span class="co">#&gt; Columns: 5</span>
<span class="co">#&gt; $ tf         &lt;chr&gt; "FOXO4", "FOXO4", "FOXO4", "FOXO4", "FOXO4", "FOXO4", "FOXO…</span>
<span class="co">#&gt; $ confidence &lt;chr&gt; "A", "A", "A", "A", "A", "A", "A", "A", "A", "A", "A", "A",…</span>
<span class="co">#&gt; $ target     &lt;chr&gt; "BCL2L11", "BCL6", "CDKN1A", "CDKN1B", "G6PC", "GADD45A", "…</span>
<span class="co">#&gt; $ mor        &lt;dbl&gt; 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,…</span>
<span class="co">#&gt; $ likelihood &lt;dbl&gt; 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,…</span>

<span class="co"># We recommend to intersect and set a minsize of targets</span>
<span class="va">network</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/intersect_regulons.html">intersect_regulons</a></span><span class="op">(</span><span class="va">mat</span>, <span class="va">network</span>, <span class="va">tf</span>, <span class="va">target</span>, minsize<span class="op">=</span><span class="fl">5</span><span class="op">)</span></code></pre></div>
</div>
<div id="how-to-decouple" class="section level2">
<h2 class="hasAnchor">
<a href="#how-to-decouple" class="anchor"></a>How to decouple?</h2>
<p>Once the data is loaded, you are one step away from achieving decoupling. This step corresponds to specifying which statistics you want to run. For more information about the defined statistics and their parameters, you can execute <code>?decouple()</code>.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/decouple.html">decouple</a></span><span class="op">(</span>
    mat <span class="op">=</span> <span class="va">mat</span>,
    network <span class="op">=</span> <span class="va">network</span>,
    .source <span class="op">=</span> <span class="st">"tf"</span>,
    .target <span class="op">=</span> <span class="st">"target"</span>,
    statistics <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"gsva"</span>, <span class="st">"wmean"</span>, <span class="st">"mlm"</span>, <span class="st">"ora"</span><span class="op">)</span>,
    args <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>
        gsva <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>,
        wmean <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>times <span class="op">=</span> <span class="fl">100</span><span class="op">)</span>,
        mlm <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>center<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>,
        ora <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>n_up<span class="op">=</span><span class="fl">300</span>, n_bottom<span class="op">=</span><span class="fl">300</span>, n_background<span class="op">=</span><span class="fl">20000</span><span class="op">)</span>
    <span class="op">)</span>
<span class="op">)</span> <span class="op"><a href="../reference/%&gt;%.html">%&gt;%</a></span>
    <span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html">glimpse</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; Rows: 140</span>
<span class="co">#&gt; Columns: 6</span>
<span class="co">#&gt; $ run_id    &lt;dbl&gt; 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, …</span>
<span class="co">#&gt; $ statistic &lt;chr&gt; "gsva", "gsva", "gsva", "gsva", "gsva", "gsva", "gsva", "gsv…</span>
<span class="co">#&gt; $ source    &lt;chr&gt; "FOXO4", "FOXO4", "FOXO4", "FOXO4", "NFIC", "NFIC", "NFIC", …</span>
<span class="co">#&gt; $ condition &lt;chr&gt; "GSM2753335", "GSM2753336", "GSM2753337", "GSM2753338", "GSM…</span>
<span class="co">#&gt; $ score     &lt;dbl&gt; -0.38043080, -0.29999174, 0.23887789, 0.09071077, -0.0845213…</span>
<span class="co">#&gt; $ p_value   &lt;dbl&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, …</span></code></pre></div>
<p>Done, you have applied different statistics to the same data set, now you can analyze them at your convenience, for example, by performing a <a href="">benchmark</a>.</p>
</div>
<div id="how-it-works" class="section level2">
<h2 class="hasAnchor">
<a href="#how-it-works" class="anchor"></a>How it works?</h2>
<div id="mapping-statistics-with-arguments" class="section level3">
<h3 class="hasAnchor">
<a href="#mapping-statistics-with-arguments" class="anchor"></a>Mapping statistics with arguments</h3>
<p>Internally, <code><a href="../reference/decouple.html">decouple()</a></code> works through <code><a href="https://purrr.tidyverse.org/reference/map2.html">purrr::map2_dfr()</a></code> to perform statistics and argument mapping. This comes with important points:</p>
<ul>
<li>
<code>statistics</code> and <code>args</code> can be unsorted vectors. <code>decouple</code> will match the given <code>statistics</code> with the given <code>args</code>, and if a statistic doesn’t have a matching <code>args</code> entry, it will run using the default parameters.</li>
<li>If an error occurs, copy the last line that was displayed with <code>show_toy_call = TRUE</code> and execute it locally. Try to fix it and correct it on the original call.</li>
</ul>
<p>See internal decouple calls and save results.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">only_stats</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/decouple.html">decouple</a></span><span class="op">(</span>
    mat <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">mat</span>, <span class="fl">5000</span><span class="op">)</span>,
    network <span class="op">=</span> <span class="va">network</span>,
    .source <span class="op">=</span> <span class="st">"tf"</span>,
    .target <span class="op">=</span> <span class="st">"target"</span>,
    statistics <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ora"</span>, <span class="st">"ulm"</span>, <span class="st">"gsva"</span><span class="op">)</span>,
    consensus_score <span class="op">=</span> <span class="cn">FALSE</span>,
    show_toy_call <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span>
<span class="co">#&gt; run_ora(mat = head(mat, 5000), network = network, .source = "tf", .target = "target")</span>
<span class="co">#&gt; run_ulm(mat = head(mat, 5000), network = network, .source = "tf", .target = "target")</span>
<span class="co">#&gt; run_gsva(mat = head(mat, 5000), network = network, .source = "tf", .target = "target")</span></code></pre></div>
<p>Run same but now with an unsorted argument.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">add_args</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/decouple.html">decouple</a></span><span class="op">(</span>
    mat <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">mat</span>, <span class="fl">5000</span><span class="op">)</span>,
    network <span class="op">=</span> <span class="va">network</span>,
    .source <span class="op">=</span> <span class="st">"tf"</span>,
    .target <span class="op">=</span> <span class="st">"target"</span>,
    statistics <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'gsva'</span>, <span class="st">'ora'</span>, <span class="st">'ulm'</span><span class="op">)</span>,
    args <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>
        ora <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>n_up<span class="op">=</span><span class="fl">300</span>, n_bottom<span class="op">=</span><span class="fl">300</span>, n_background<span class="op">=</span><span class="fl">20000</span><span class="op">)</span>,
        ulm <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>center <span class="op">=</span> <span class="cn">F</span><span class="op">)</span>
    <span class="op">)</span>,
    consensus_score <span class="op">=</span> <span class="cn">FALSE</span>,
    show_toy_call <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span>
<span class="co">#&gt; run_ora(mat = head(mat, 5000), network = network, .source = "tf", .target = "target",</span>
<span class="co">#&gt; * n_up = 300, n_bottom = 300, n_background = 20000)</span>
<span class="co">#&gt; run_ulm(mat = head(mat, 5000), network = network, .source = "tf", .target = "target",</span>
<span class="co">#&gt; * center = FALSE)</span>
<span class="co">#&gt; run_gsva(mat = head(mat, 5000), network = network, .source = "tf", .target = "target")</span></code></pre></div>
<p>Now compare results and see there is not difference.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/all.equal.html">all.equal</a></span><span class="op">(</span><span class="va">only_stats</span>, <span class="va">add_args</span><span class="op">)</span>
<span class="co">#&gt; [1] TRUE</span></code></pre></div>
</div>
<div id="mapping-network-columns" class="section level3">
<h3 class="hasAnchor">
<a href="#mapping-network-columns" class="anchor"></a>Mapping network columns</h3>
<p>To carry out the column mapping, <code>decoupleR</code> relies on the selection provided by the <em><a href="https://github.com/tidyverse/tidyselect">tidyselect</a></em> package. Some of the selection it provides are:</p>
<ul>
<li>Symbols</li>
<li>Strings</li>
<li>Position</li>
</ul>
<p>Let’s see an example. Input network has the following columns:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">network</span> <span class="op"><a href="../reference/%&gt;%.html">%&gt;%</a></span>
    <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; [1] "tf"         "confidence" "target"     "mor"        "likelihood"</span></code></pre></div>
<p>We can use the way we like to do the mapping, even a combination of ways to do it. This applies not only to the decouple function, but to all functions of the <code>decoupleR statistics</code> family, identifiable by the <code>run_</code> prefix.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">this_column</span> <span class="op">&lt;-</span> <span class="st">"target"</span>
<span class="va">viper_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/decouple.html">decouple</a></span><span class="op">(</span>
    mat <span class="op">=</span> <span class="va">mat</span>,
    network <span class="op">=</span> <span class="va">network</span>,
    .source <span class="op">=</span> <span class="va">tf</span>,
    .target <span class="op">=</span> <span class="op">!</span><span class="op">!</span><span class="va">this_column</span>,
    statistics <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"viper"</span><span class="op">)</span>,
    args <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>
        viper <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>
            .mor <span class="op">=</span> <span class="fl">4</span>,
            .likelihood <span class="op">=</span> <span class="st">"likelihood"</span>,
            verbose <span class="op">=</span> <span class="cn">FALSE</span>
        <span class="op">)</span>
    <span class="op">)</span>,
    show_toy_call <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span>
<span class="co">#&gt; run_viper(mat = mat, network = network, .source = tf, .target = "target",</span>
<span class="co">#&gt; *   .mor = 4, .likelihood = "likelihood", verbose = FALSE)</span></code></pre></div>
</div>
</div>
</div>
<div id="reproducibility" class="section level1">
<h1 class="hasAnchor">
<a href="#reproducibility" class="anchor"></a>Reproducibility</h1>
<div id="special-thanks" class="section level2">
<h2 class="hasAnchor">
<a href="#special-thanks" class="anchor"></a>Special thanks</h2>
<p>The <em><a href="https://bioconductor.org/packages/3.13/decouopleR">decouopleR</a></em> package (Badia-i-Mompel, Vélez, Braunger, Geiss, Dimitrov, Müller-Dott, Taus, Dugourd, Holland, Flores, and Saez-Rodriguez, 2021) was made possible thanks to:</p>
<ul>
<li>R (R Core Team, 2021)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.13/BiocStyle">BiocStyle</a></em> (Oleś, 2021)</li>
<li><em><a href="https://CRAN.R-project.org/package=knitcitations">knitcitations</a></em></li>
<li>
<em><a href="https://CRAN.R-project.org/package=knitr">knitr</a></em> (Xie, 2021)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=rmarkdown">rmarkdown</a></em> (Allaire, Xie, McPherson, Luraschi, Ushey, Atkins, Wickham, Cheng, Chang, and Iannone, 2021)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=sessioninfo">sessioninfo</a></em> (Wickham, Chang, Flight, Müller, and Hester, 2021)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=testthat">testthat</a></em> (Wickham, 2011)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.13/GSVA">GSVA</a></em> (H{ä}nzelmann, Castelo, and Guinney, 2013)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.13/viper">viper</a></em> (Alvarez, Shen, Giorgi, Lachmann, Ding, Ye, and Califano, 2016)</li>
</ul>
<p>This package was developed using <em><a href="https://github.com/lcolladotor/biocthis">biocthis</a></em>.</p>
</div>
<div id="vignette" class="section level2">
<h2 class="hasAnchor">
<a href="#vignette" class="anchor"></a>Vignette</h2>
<div id="create" class="section level3">
<h3 class="hasAnchor">
<a href="#create" class="anchor"></a>Create</h3>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Create the vignette</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/rstudio/rmarkdown">rmarkdown</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/system.time.html">system.time</a></span><span class="op">(</span><span class="fu"><a href="https://pkgs.rstudio.com/rmarkdown/reference/render.html">render</a></span><span class="op">(</span><span class="st">"decoupleR.Rmd"</span>, <span class="st">"BiocStyle::html_document"</span><span class="op">)</span><span class="op">)</span>

<span class="co"># Extract the R code</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yihui.org/knitr/">knitr</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/knitr/man/knit.html">knit</a></span><span class="op">(</span><span class="st">"decoupleR.Rmd"</span>, tangle <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
</div>
<div id="wallclock-time-spent-generating-the-vignette" class="section level3">
<h3 class="hasAnchor">
<a href="#wallclock-time-spent-generating-the-vignette" class="anchor"></a>Wallclock time spent generating the vignette</h3>
<pre><code>#&gt; Time difference of 14.092 secs</code></pre>
</div>
</div>
<div id="session-information" class="section level2">
<h2 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session information</h2>
<pre><code>#&gt; ─ Session info  ──────────────────────────────────────────────────────────────────────────────────────────────────────
#&gt;  hash: pine decoration, vulcan salute: dark skin tone, ox
#&gt; 
#&gt;  setting  value
#&gt;  version  R version 4.1.1 (2021-08-10)
#&gt;  os       Ubuntu 20.04.3 LTS
#&gt;  system   x86_64, linux-gnu
#&gt;  ui       X11
#&gt;  language (EN)
#&gt;  collate  en_US.UTF-8
#&gt;  ctype    en_US.UTF-8
#&gt;  tz       UTC
#&gt;  date     2021-11-12
#&gt;  pandoc   2.11.4 @ /usr/local/bin/ (via rmarkdown)
#&gt; 
#&gt; ─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────
#&gt;  package              * version  date (UTC) lib source
#&gt;  annotate               1.70.0   2021-05-19 [1] Bioconductor
#&gt;  AnnotationDbi          1.54.1   2021-06-08 [1] Bioconductor
#&gt;  backports              1.3.0    2021-10-27 [1] RSPM (R 4.1.0)
#&gt;  beachmat               2.8.1    2021-08-12 [1] Bioconductor
#&gt;  Biobase                2.52.0   2021-05-19 [1] Bioconductor
#&gt;  BiocGenerics           0.38.0   2021-05-19 [1] Bioconductor
#&gt;  BiocManager            1.30.16  2021-06-15 [2] CRAN (R 4.1.1)
#&gt;  BiocParallel           1.26.2   2021-08-22 [1] Bioconductor
#&gt;  BiocSingular           1.8.1    2021-06-08 [1] Bioconductor
#&gt;  BiocStyle            * 2.20.2   2021-06-17 [1] Bioconductor
#&gt;  Biostrings             2.60.2   2021-08-05 [1] Bioconductor
#&gt;  bit                    4.0.4    2020-08-04 [1] RSPM (R 4.1.0)
#&gt;  bit64                  4.0.5    2020-08-30 [1] RSPM (R 4.1.0)
#&gt;  bitops                 1.0-7    2021-04-24 [1] RSPM (R 4.1.0)
#&gt;  blob                   1.2.2    2021-07-23 [1] RSPM (R 4.1.0)
#&gt;  bookdown               0.24     2021-09-02 [1] RSPM (R 4.1.0)
#&gt;  broom                  0.7.10   2021-10-31 [1] RSPM (R 4.1.0)
#&gt;  bslib                  0.3.1    2021-10-06 [1] RSPM (R 4.1.0)
#&gt;  cachem                 1.0.6    2021-08-19 [2] RSPM (R 4.1.0)
#&gt;  class                  7.3-19   2021-05-03 [3] CRAN (R 4.1.1)
#&gt;  cli                    3.1.0    2021-10-27 [2] RSPM (R 4.1.0)
#&gt;  crayon                 1.4.2    2021-10-29 [2] RSPM (R 4.1.0)
#&gt;  DBI                    1.1.1    2021-01-15 [1] RSPM (R 4.1.0)
#&gt;  decoupleR            * 2.1.2    2021-11-12 [1] Bioconductor
#&gt;  DelayedArray           0.18.0   2021-05-19 [1] Bioconductor
#&gt;  DelayedMatrixStats     1.14.3   2021-08-26 [1] Bioconductor
#&gt;  desc                   1.4.0    2021-09-28 [2] RSPM (R 4.1.0)
#&gt;  digest                 0.6.28   2021-09-23 [2] RSPM (R 4.1.0)
#&gt;  dplyr                * 1.0.7    2021-06-18 [2] RSPM (R 4.1.0)
#&gt;  e1071                  1.7-9    2021-09-16 [1] RSPM (R 4.1.0)
#&gt;  ellipsis               0.3.2    2021-04-29 [2] RSPM (R 4.1.0)
#&gt;  evaluate               0.14     2019-05-28 [2] RSPM (R 4.1.0)
#&gt;  fansi                  0.5.0    2021-05-25 [2] RSPM (R 4.1.0)
#&gt;  fastmap                1.1.0    2021-01-25 [2] RSPM (R 4.1.0)
#&gt;  fs                     1.5.0    2020-07-31 [2] RSPM (R 4.1.0)
#&gt;  generics               0.1.1    2021-10-25 [2] RSPM (R 4.1.0)
#&gt;  GenomeInfoDb           1.28.4   2021-09-05 [1] Bioconductor
#&gt;  GenomeInfoDbData       1.2.6    2021-11-05 [1] Bioconductor
#&gt;  GenomicRanges          1.44.0   2021-05-19 [1] Bioconductor
#&gt;  glue                   1.5.0    2021-11-07 [2] RSPM (R 4.1.0)
#&gt;  graph                  1.70.0   2021-05-19 [1] Bioconductor
#&gt;  GSEABase               1.54.0   2021-05-19 [1] Bioconductor
#&gt;  GSVA                   1.40.1   2021-06-06 [1] Bioconductor
#&gt;  HDF5Array              1.20.0   2021-05-19 [1] Bioconductor
#&gt;  htmltools              0.5.2    2021-08-25 [1] RSPM (R 4.1.0)
#&gt;  httr                   1.4.2    2020-07-20 [2] RSPM (R 4.1.0)
#&gt;  IRanges                2.26.0   2021-05-19 [1] Bioconductor
#&gt;  irlba                  2.3.3    2019-02-05 [1] RSPM (R 4.1.0)
#&gt;  jquerylib              0.1.4    2021-04-26 [1] RSPM (R 4.1.0)
#&gt;  jsonlite               1.7.2    2020-12-09 [2] RSPM (R 4.1.0)
#&gt;  KEGGREST               1.32.0   2021-05-19 [1] Bioconductor
#&gt;  kernlab                0.9-29   2019-11-12 [1] RSPM (R 4.1.0)
#&gt;  KernSmooth             2.23-20  2021-05-03 [3] CRAN (R 4.1.1)
#&gt;  knitr                  1.36     2021-09-29 [2] RSPM (R 4.1.0)
#&gt;  lattice                0.20-45  2021-09-22 [3] RSPM (R 4.1.0)
#&gt;  lifecycle              1.0.1    2021-09-24 [2] RSPM (R 4.1.0)
#&gt;  lubridate              1.8.0    2021-10-07 [1] RSPM (R 4.1.0)
#&gt;  magrittr               2.0.1    2020-11-17 [2] RSPM (R 4.1.0)
#&gt;  MASS                   7.3-54   2021-05-03 [3] CRAN (R 4.1.1)
#&gt;  Matrix                 1.3-4    2021-06-01 [3] CRAN (R 4.1.1)
#&gt;  MatrixGenerics         1.4.3    2021-08-26 [1] Bioconductor
#&gt;  matrixStats            0.61.0   2021-09-17 [1] RSPM (R 4.1.0)
#&gt;  memoise                2.0.0    2021-01-26 [2] RSPM (R 4.1.0)
#&gt;  mixtools               1.2.0    2020-02-07 [1] RSPM (R 4.1.0)
#&gt;  pillar                 1.6.4    2021-10-18 [2] RSPM (R 4.1.0)
#&gt;  pkgconfig              2.0.3    2019-09-22 [2] RSPM (R 4.1.0)
#&gt;  pkgdown                1.6.1    2020-09-12 [1] RSPM (R 4.1.0)
#&gt;  plyr                   1.8.6    2020-03-03 [1] RSPM (R 4.1.0)
#&gt;  png                    0.1-7    2013-12-03 [1] RSPM (R 4.1.0)
#&gt;  proxy                  0.4-26   2021-06-07 [1] RSPM (R 4.1.0)
#&gt;  purrr                  0.3.4    2020-04-17 [2] RSPM (R 4.1.0)
#&gt;  R6                     2.5.1    2021-08-19 [2] RSPM (R 4.1.0)
#&gt;  ragg                   1.2.0    2021-10-30 [1] RSPM (R 4.1.0)
#&gt;  ranger                 0.13.1   2021-07-14 [1] RSPM (R 4.1.0)
#&gt;  Rcpp                   1.0.7    2021-07-07 [1] RSPM (R 4.1.0)
#&gt;  RCurl                  1.98-1.5 2021-09-17 [1] RSPM (R 4.1.0)
#&gt;  RefManageR           * 1.3.0    2020-11-13 [1] RSPM (R 4.1.0)
#&gt;  rhdf5                  2.36.0   2021-05-19 [1] Bioconductor
#&gt;  rhdf5filters           1.4.0    2021-05-19 [1] Bioconductor
#&gt;  Rhdf5lib               1.14.2   2021-07-06 [1] Bioconductor
#&gt;  rlang                  0.4.12   2021-10-18 [2] RSPM (R 4.1.0)
#&gt;  rmarkdown              2.11     2021-09-14 [1] RSPM (R 4.1.0)
#&gt;  RobustRankAggreg       1.1      2013-06-03 [1] RSPM (R 4.1.0)
#&gt;  rpart                  4.1-15   2019-04-12 [3] CRAN (R 4.1.1)
#&gt;  rprojroot              2.0.2    2020-11-15 [2] RSPM (R 4.1.0)
#&gt;  RSQLite                2.2.8    2021-08-21 [1] RSPM (R 4.1.0)
#&gt;  rsvd                   1.0.5    2021-04-16 [1] RSPM (R 4.1.0)
#&gt;  S4Vectors              0.30.2   2021-10-03 [1] Bioconductor
#&gt;  sass                   0.4.0    2021-05-12 [1] RSPM (R 4.1.0)
#&gt;  ScaledMatrix           1.0.0    2021-05-19 [1] Bioconductor
#&gt;  segmented              1.3-4    2021-04-22 [1] RSPM (R 4.1.0)
#&gt;  sessioninfo            1.2.1    2021-11-02 [2] RSPM (R 4.1.0)
#&gt;  SingleCellExperiment   1.14.1   2021-05-21 [1] Bioconductor
#&gt;  sparseMatrixStats      1.4.2    2021-08-08 [1] Bioconductor
#&gt;  speedglm               0.3-3    2021-01-08 [1] RSPM (R 4.1.0)
#&gt;  stringi                1.7.5    2021-10-04 [2] RSPM (R 4.1.0)
#&gt;  stringr                1.4.0    2019-02-10 [2] RSPM (R 4.1.0)
#&gt;  SummarizedExperiment   1.22.0   2021-05-19 [1] Bioconductor
#&gt;  survival               3.2-13   2021-08-24 [3] RSPM (R 4.1.0)
#&gt;  systemfonts            1.0.3    2021-10-13 [1] RSPM (R 4.1.0)
#&gt;  textshaping            0.3.6    2021-10-13 [1] RSPM (R 4.1.0)
#&gt;  tibble                 3.1.6    2021-11-07 [2] RSPM (R 4.1.0)
#&gt;  tidyr                  1.1.4    2021-09-27 [2] RSPM (R 4.1.0)
#&gt;  tidyselect             1.1.1    2021-04-30 [2] RSPM (R 4.1.0)
#&gt;  utf8                   1.2.2    2021-07-24 [2] RSPM (R 4.1.0)
#&gt;  vctrs                  0.3.8    2021-04-29 [2] RSPM (R 4.1.0)
#&gt;  viper                  1.26.0   2021-05-19 [1] Bioconductor
#&gt;  withr                  2.4.2    2021-04-18 [2] RSPM (R 4.1.0)
#&gt;  xfun                   0.28     2021-11-04 [2] RSPM (R 4.1.0)
#&gt;  XML                    3.99-0.8 2021-09-17 [1] RSPM (R 4.1.0)
#&gt;  xml2                   1.3.2    2020-04-23 [2] RSPM (R 4.1.0)
#&gt;  xtable                 1.8-6    2020-06-19 [1] R-Forge (R 4.1.1)
#&gt;  XVector                0.32.0   2021-05-19 [1] Bioconductor
#&gt;  yaml                   2.2.1    2020-02-01 [2] RSPM (R 4.1.0)
#&gt;  zlibbioc               1.38.0   2021-05-19 [1] Bioconductor
#&gt; 
#&gt;  [1] /__w/_temp/Library
#&gt;  [2] /usr/local/lib/R/site-library
#&gt;  [3] /usr/local/lib/R/library
#&gt; 
#&gt; ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</code></pre>
</div>
</div>
<div id="bibliography" class="section level1">
<h1 class="hasAnchor">
<a href="#bibliography" class="anchor"></a>Bibliography</h1>
<p>This vignette was generated using <em><a href="https://bioconductor.org/packages/3.13/BiocStyle">BiocStyle</a></em> (Oleś, 2021) with <em><a href="https://CRAN.R-project.org/package=knitr">knitr</a></em> (Xie, 2021) and <em><a href="https://CRAN.R-project.org/package=rmarkdown">rmarkdown</a></em> (Allaire, Xie, McPherson, et al., 2021) running behind the scenes.</p>
<p>Citations made with <em><a href="https://CRAN.R-project.org/package=RefManageR">RefManageR</a></em> (McLean, 2017).</p>
<p>
<a id="bib-allaire2021rmarkdown"></a><a href="#cite-allaire2021rmarkdown">[1]</a><cite> J. Allaire, Y. Xie, J. McPherson, et al. <em>rmarkdown: Dynamic Documents for R</em>. R package version 2.11. 2021. URL: <a href="https://github.com/rstudio/rmarkdown">https://github.com/rstudio/rmarkdown</a>.</cite>
</p>
<p>
<a id="bib-alvarez2016functional"></a><a href="#cite-alvarez2016functional">[2]</a><cite> M. J. Alvarez, Y. Shen, F. M. Giorgi, et al. “Functional characterization of somatic mutations in cancer using network-based inference of protein activity”. In: <em>Nature genetics</em> 48.8 (2016), pp. 838–47.</cite>
</p>
<p>
<a id="bib-badiaimompel2021decoupler"></a><a href="#cite-badiaimompel2021decoupler">[3]</a><cite> P. Badia-i-Mompel, J. Vélez, J. Braunger, et al. “decoupleR: Ensemble of computational methods to infer biological activities from omics data”. In: <em>bioRxiv</em> (2021). DOI: <a href="https://doi.org/https://doi.org/10.1101/2021.11.04.467271">https://doi.org/10.1101/2021.11.04.467271</a>.</cite>
</p>
<p>
<a id="bib-hanzelmann2013gsva"></a><a href="#cite-hanzelmann2013gsva">[4]</a><cite> S. Hänzelmann, R. Castelo, and J. Guinney. “GSVA: gene set variation analysis for microarray and RNA-Seq data”. In: <em>BMC Bioinformatics</em> 14 (2013), p. 7. DOI: <a href="https://doi.org/10.1186/1471-2105-14-7">10.1186/1471-2105-14-7</a>. URL: <a href="http://www.biomedcentral.com/1471-2105/14/7">http://www.biomedcentral.com/1471-2105/14/7</a>.</cite>
</p>
<p>
<a id="bib-mclean2017refmanager"></a><a href="#cite-mclean2017refmanager">[5]</a><cite> M. W. McLean. “RefManageR: Import and Manage BibTeX and BibLaTeX References in R”. In: <em>The Journal of Open Source Software</em> (2017). DOI: <a href="https://doi.org/10.21105/joss.00338">10.21105/joss.00338</a>.</cite>
</p>
<p>
<a id="bib-ole2021biocstyle"></a><a href="#cite-ole2021biocstyle">[6]</a><cite> A. Oleś. <em>BiocStyle: Standard styles for vignettes and other Bioconductor documents</em>. R package version 2.20.2. 2021. URL: <a href="https://github.com/Bioconductor/BiocStyle">https://github.com/Bioconductor/BiocStyle</a>.</cite>
</p>
<p>
<a id="bib-2021language"></a><a href="#cite-2021language">[7]</a><cite> R Core Team. <em>R: A Language and Environment for Statistical Computing</em>. R Foundation for Statistical Computing. Vienna, Austria, 2021. URL: <a href="https://www.R-project.org/">https://www.R-project.org/</a>.</cite>
</p>
<p>
<a id="bib-wickham2011testthat"></a><a href="#cite-wickham2011testthat">[8]</a><cite> H. Wickham. “testthat: Get Started with Testing”. In: <em>The R Journal</em> 3 (2011), pp. 5–10. URL: <a href="https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf">https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf</a>.</cite>
</p>
<p>
<a id="bib-wickham2021sessioninfo"></a><a href="#cite-wickham2021sessioninfo">[9]</a><cite> H. Wickham, W. Chang, R. Flight, et al. <em>sessioninfo: R Session Information</em>. <a href="https://github.com/r-lib/sessioninfo#readme" class="uri">https://github.com/r-lib/sessioninfo#readme</a>, <a href="https://r-lib.github.io/sessioninfo/" class="uri">https://r-lib.github.io/sessioninfo/</a>. 2021.</cite>
</p>
<p>
<a id="bib-xie2021knitr"></a><a href="#cite-xie2021knitr">[10]</a><cite> Y. Xie. <em>knitr: A General-Purpose Package for Dynamic Report Generation in R</em>. R package version 1.36. 2021. URL: <a href="https://yihui.org/knitr/">https://yihui.org/knitr/</a>.</cite>
</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Pau Badia-i-Mompel, Jesús Vélez, Jana Braunger, Celina Geiss, Daniel Dimitrov, Sophia Müller-Dott, Petr Taus, Aurélien Dugourd, Christian H. Holland, Ricardo O. Ramirez Flores, Julio Saez-Rodriguez.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
